app.go

```diff
@@ -74,6 +84,10 @@ var (
 // Provide constructors into the D.I. Container, their types will be available
 // to all other constructors, and called lazily at startup
 func (s *App) Provide(constructors ...interface{}) {
+	if s.state != stopped {
+		s.logger.Panic(errors.New("Provide can only be called before an App is started"))
```

Comment:
I don't quite understand panic over errors, you simply don't want to change the function signature?